pkg/mysql/conn.go

```diff
@@ -1034,3 +1064,37 @@ func (c *Conn) Close() {
 func (c *Conn) IsClosed() bool {
 	return c.closed.Get()
 }
+
+func connCheck(conn net.Conn) error {
+	var sysErr error
+
+	sysConn, ok := conn.(syscall.Conn)
+	if !ok {
+		return nil
```

Comment:
better to return error instead of nil?

Reply:
Only `syscall.Conn` `SyscallConn()` func return `RawConn`, we use `RawConn` to check connection alive. If a connection is't `syscall.Conn` does't mean it can't read.